### PR TITLE
[#1112] Allow to limit the size of whole request or the size of one uploaded file in a request

### PR DESCRIPTION
**Pull request by cbxp**
…file in a request.

 The configuration parameters are: 'upload.sizeMax' and 'upload.fileSizeMax'.
 The defaults stay the same -1 which means 'unlimited'.

### CODE REVIEW

**Comment by asolntsev** [comment]
@xael-fry This is definitely good PR. I suggest to merge it ASAP.

**Comment by cbxp** [comment]
@xael-fry We have updated the documentation.
By the way, I suggest to put some reasonable default values. Say, 
```
upload.maxFileSize=10240   (10 mb)
upload.maxRequestSize=30720  (30 mb)
```

Without that, it's possible that hacker can upload 100TB file to server.

File: framework/src/play/data/parsing/ApacheMultipartParser.java [hunk 1]
```diff
@@ -612,12 +612,12 @@ public void setHeaders(FileItemHeaders pHeaders) {
      * The maximum size permitted for the complete request, as opposed to
      * {@link #fileSizeMax}. A value of -1 indicates no maximum.
```
Comment (xael-fry) [inline framework/src/play/data/parsing/ApacheMultipartParser.java]: fileSizeMax no longer exist as you rename it

**Comment by asolntsev** [comment]
@xael-fry Fixed parameter name in javadoc. What about setting default values to something reasonable?

**Comment by xael-fry** [comment]
@asolntsev I don't think we should define default value here, some user handle request size limit with a a front apache server, so setting new value may cause conflict when upgrading

**Pull request merged**